parsimonious/tests/test_grammar.py

```diff
@@ -378,6 +392,26 @@ def test_unconnected_custom_rules(self):
         eq_(grammar.parse(s),
             Node(grammar['one_char'], s, 0, 1))
 
+    def test_callability_of_routines(self):
+        ok_(is_callable(function_rule))
+        ok_(is_callable(self.method_rule))
+        ok_(is_callable(self.descriptor_rule))
+
+    def test_callability_custom_rules(self):
+        """Confirms that functions, methods and method descriptors can all be
+        used to supply custom grammar rules.
+        """
+        grammar = Grammar("""
+            default = function method descriptor
+            """,
+            function=function_rule,
+            method=self.method_rule,
+            descriptor=self.descriptor_rule,
```

Comment:
I don't think this is testing the method descriptor case correctly. When you reference the static method with `self.` it seems to bind it into a function. To properly test this I would wrap `descriptor_rule` in an object per your original example.

```
@staticmethod
def descriptor_rule(text, pos):
    ...

rules = {"descriptor": descriptor_rule}
```

Then in your test you would do `descriptor=self.rules["descriptor"]`

Reply:
Nice catch, thanks @evanh!  That should now be handled correctly, I think.  It's been a while since I got into the weeds on the differences between methods and descriptors, so please do double-check me on the fix applied here.

Reply:
@evanh Very good catch—and well-described!

And thanks for the great patch, @jayaddison; this is definitely a hole we should fill!